### ISSUE TITLE
Unable to send file

### ISSUE THREAD

**Issue by epsilontheta** [dino/dino#752]
When I try to send a file I get this error in the debug log output:

```
(dino:59972): dconf-DEBUG: 23:23:14.277: change_notify: /org/gtk/settings/file-chooser/
(dino:59972): Gtk-DEBUG: 23:23:14.289: State 0 for GtkLabel 0x562b9a816dd0 doesn't match state 128 set via gtk_style_context_set_state ()
(dino:59972): OMEMO-DEBUG: 23:23:14.291: file_encryptor.vala:47: Encrypting file xxx.jpeg as xxx.jpeg

(dino:59972): libdino-WARNING **: 23:23:14.291: file_manager.vala:118: Send file error: Cannot upload http file meta over Jingle
```

Which standards can dino use for file transfers?
What are things that I can try to improve it?

I can transfer files using gajim, not very fast though.

Used version: dino 0.1.0.

**Comment by licaon-kter** [comment]
Your server needs XEP-0363 HTTP Upload support.

**Comment by vandenoever** [comment]
I noticed that file transfer fails with OMEMO even when it works without encryption.

**Comment by mss-cyclist** [comment]
Same here. Version 0.1.0 cannot send files when Omemo is enabled. Files sent with Omemo cannot be received either.
Disabling Omemo enables file sending and receiving.

`(dino:55373): libdino-WARNING **: 15:36:02.644: file_manager.vala:118: Send file error: HTTP upload error: Nicht akzeptables TLS-Zertifikat`

After re-enabling Omemo send and receive of files is working again.

**Comment by licaon-kter** [comment]
The error says something about a certificate, that's not important? Did you check that already?

**Comment by mss-cyclist** [comment]
Yeah. I checked that. Certificate is ok. This does not take away that aparently Omemo does have a problem with the first file if it was sent/received encrypted. Sending one file unencrypted solves the issue. From that point on Omemo transfer for files in both direction works.

**Comment by licaon-kter** [comment]
So it's a key issue, not exchanged yet?

**Comment by mss-cyclist** [comment]
To be honest: at first there was a certificate issue. The certificate for the upload subdomain pointed to the wrong certificate. This was resolved but the Dino did have the symptoms mentioned before.
Just did a test with another user on my server. He was not able to send and receive as well. But as of today he can. Without disabling and re-enabling Omemo.
So atm I cannot say if my Dino blocked both Dino's or if it was the certificate. Btw: the certificate did not affect other clients.
Atm I can not say if there was something wrong with my local Dino which blocked both Dino's.

**Comment by ericschdt** [comment]
```
file_manager.vala:125: Send file error: HTTP upload error: Invalid TLS certificate
```

I have this error as well, however _only_ on Dino w/ or w/o OMEMO, while Conversations just works fine. What I would require here is some information about the certificate:
Which domain does Dino tries to upload, which certificate is used? Is the certificate for a different domain, is it expired, etc. ? 
Right now, it's hard to figure out, what the issue is.

**Comment by licaon-kter** [comment]
Domain for uploads is decided by the server setup.
